diff --git a/node_modules/solid-js/dist/server.cjs b/node_modules/solid-js/dist/server.cjs
index 78cce5e..9ca5970 100644
--- a/node_modules/solid-js/dist/server.cjs
+++ b/node_modules/solid-js/dist/server.cjs
@@ -430,10 +430,18 @@ function createResource(source, fetcher, options = {}) {
 }
 function lazy(fn) {
   let resolved;
-  const p = fn();
+  let p;
+  function load() {
+    if (!p) {
+      p = fn();
+      p.then(mod => resolved = mod.default);
+
+    }
+    return p;
+  }
   const contexts = new Set();
-  p.then(mod => resolved = mod.default);
   const wrap = props => {
+    load();
     const id = sharedConfig.context.id.slice(0, -1);
     if (resolved) return resolved(props);
     const ctx = useContext(SuspenseContext);
@@ -451,7 +459,7 @@ function lazy(fn) {
     });
     return "";
   };
-  wrap.preload = () => p;
+  wrap.preload = load;
   return wrap;
 }
 function suspenseComplete(c) {
